File: Simulacion_JuanAntonio/Unidad_2/R Studio Projects/GeneracióndeNúmerosPseudoaleatorios.R
fibonacci_aditivo <- function(seed, n) {
  m <- 2^31 - 1
  x <- numeric(n+2)
  x[1] <- seed
  x[2] <- (seed * 37) %% m
  
  for (i in 3:(n+2)) {
    x[i] <- (x[i-1] + x[i-2]) %% m
  }
  return(x[3:(n+2)])
}

cat("Generación de números pseudoaleatorios con método aditivo (R):\n")
print(fibonacci_aditivo(12345, 10))

File: Simulacion_JuanAntonio/Unidad_2/Python Projects/GeneraciónDeNúmerosPseudoaleatorios.py
def cuadrados_medios(seed, n):
    numeros = []
    x = seed
    for _ in range(n):
        x = (x * x) // 100 % 10000  # cuadrado, quitar extremos
        numeros.append(x)
    return numeros

print("Generación de números pseudoaleatorios con el método del cuadrado medio:")
print(cuadrados_medios(5735, 10))
